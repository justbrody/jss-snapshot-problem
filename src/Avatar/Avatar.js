import React from 'react'
import injectSheet from 'react-jss'
import clsx from 'clsx'

export const styles = theme => ({
  root: {
    position: 'relative',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexShrink: 0,
    width: 40,
    height: 40,
    fontFamily: theme.typography.fontFamily,
    fontSize: theme.typography.pxToRem(20),
    borderRadius: '50%',
    overflow: 'hidden',
    userSelect: 'none'
  },
  colorDefault: {
    color: theme.palette.background.default,
    backgroundColor:
      theme.palette.type === 'light'
        ? theme.palette.grey[400]
        : theme.palette.grey[600]
  },
  img: {
    width: '100%',
    height: '100%',
    textAlign: 'center',
    objectFit: 'cover'
  }
})

const Avatar = React.forwardRef(function Avatar (props, ref) {
  const {
    alt,
    children: childrenProp,
    childrenClassName: childrenClassNameProp,
    classes,
    className: classNameProp,
    component: Component = 'div',
    imgProps,
    sizes,
    src,
    srcSet,
    ...other
  } = props

  let children = null
  const img = src || srcSet

  if (img) {
    children = (
      <img
        alt={alt}
        src={src}
        srcSet={srcSet}
        sizes={sizes}
        className={classes.img}
        {...imgProps}
      />
    )
  } else if (childrenClassNameProp && React.isValidElement(childrenProp)) {
    children = React.cloneElement(childrenProp, {
      className: clsx(childrenClassNameProp, childrenProp.props.className)
    })
  } else {
    children = childrenProp
  }

  return (
    <Component
      className={clsx(
        classes.root,
        classes.system,
        {
          [classes.colorDefault]: !img
        },
        classNameProp
      )}
      ref={ref}
      {...other}
    >
      {children}
    </Component>
  )
})

export default injectSheet(styles)(Avatar)
