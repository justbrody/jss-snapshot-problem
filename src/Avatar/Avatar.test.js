import React from 'react'
import JssProvider from 'react-jss/lib/JssProvider'
import { ThemeProvider, SheetsRegistry } from 'react-jss'
import renderer from 'react-test-renderer'

import Avatar from './Avatar'

it('matches css snapshot', () => {
  const sheets = new SheetsRegistry()

  renderer
    .create(
      <JssProvider registry={sheets}>
        <ThemeProvider
          theme={{
            typography: {
              fontFamily: 'Times New Roman, Arial',
              pxToRem: x => x * 2
            },
            palette: {
              background: { default: '#000000' },
              grey: []
            }
          }}
        >
          <Avatar src='/..' />
        </ThemeProvider>
      </JssProvider>
    )
    .toJSON()

  expect(sheets.toString()).toMatchSnapshot()
})
