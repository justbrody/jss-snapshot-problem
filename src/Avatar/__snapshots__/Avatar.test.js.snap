// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches css snapshot 1`] = `
".Component-root-0-1-1 {
  width: 40px;
  height: 40px;
  display: flex;
  position: relative;
  overflow: hidden;
  font-size: 40px;
  flex-shrink: 0;
  font-family: Times New Roman, Arial;
  align-items: center;
  user-select: none;
  border-radius: 50%;
  justify-content: center;
}
.Component-colorDefault-0-1-2 {
  color: #000000;
}
.Component-img-0-1-3 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
}"
`;
